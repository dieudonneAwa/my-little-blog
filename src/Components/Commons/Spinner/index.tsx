import styled, { keyframes } from 'styled-components';
import { space } from 'styled-system';
import { DefaultTheme } from '../../../Theme';

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

const StyledSpinner = styled.span<{
  size: string;
  color: string;
  ml: any;
  theme: DefaultTheme;
}>`
  &::before {
    content: '';
    -webkit-transform-origin: center;
    animation: ${rotate} 500ms infinite linear;
    border-radius: 50%;
    border: 0.2rem solid ${(p) => p.theme.colors.emerald};
    border-right-color: transparent;
    border-top-color: transparent;
    content: '';
    display: inline-block;
    height: ${(props) => props.size};
    transform-origin: center;
    width: ${(props) => props.size};

    // Styled System Props
    ${space}
  }
`;

export const SpinnerContainer = styled.div<{ height: string }>`
  min-height: ${(props) => (props.height ? props.height : '100%')};
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export interface Props {
  /**
   * Color of the spinner
   */
  color: string;
  /**
   * Size of the spinner
   */
  size: string;
}

export const Spinner = ({ color, size, ...rest }: Props | any) => (
  // @ts-ignore
  <StyledSpinner color={color} size={size} {...rest} />
);

export default Spinner;

Spinner.defaultProps = {
  color: 'c300',
  size: '1rem',
};
