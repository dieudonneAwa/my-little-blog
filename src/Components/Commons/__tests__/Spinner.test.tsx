/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Spinner from '../Spinner';

describe('Spinner', (): void => {
  let wrapper: ShallowWrapper<void>;
  beforeEach((): void => {
    wrapper = shallow(<Spinner />);
  });

  it('should render Spinner component', (): void => {
    expect(wrapper).toMatchSnapshot();
  });
});
