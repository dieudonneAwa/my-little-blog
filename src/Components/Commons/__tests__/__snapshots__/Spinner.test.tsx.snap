// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render Spinner component 1`] = `
<styled.span
  color="c300"
  size="1rem"
/>
`;
